vpc/tool/container/setup_container_linux.go

```diff
@@ -292,31 +324,38 @@ func ipaddressToHandle(ip net.IP) uint16 {
 	return binary.BigEndian.Uint16([]byte(ip.To4()[2:4]))
 }
 
-func teardownNetwork(ctx context.Context, allocation types.LegacyAllocation, link netlink.Link, netnsfd int) {
-	deleteLink(ctx, link, netnsfd)
-	ip := net.ParseIP(allocation.IPV4Address.Address.Address)
-
-	// Removing the classes automatically removes the qdiscs
-	ifbEgress, err := netlink.LinkByName(vpc.EgressIFB)
-	if err == nil {
-		removeClass(ctx, ip, ifbEgress)
-	} else {
-		logger.G(ctx).WithError(err).Warning("Unable to find ifb egress, during deallocation")
-	}
+func teardownNetwork(ctx context.Context, allocation types.LegacyAllocation, links []netlink.Link, netnsfd []interface{}) {
+	for i, link := range links {
+		deleteLink(ctx, link, netnsfd[i])
+		ip := net.ParseIP(allocation.IPV4Address.Address.Address)
+
+		// Removing the classes automatically removes the qdiscs
+		ifbEgress, err := netlink.LinkByName(vpc.EgressIFB)
+		if err == nil {
+			removeClass(ctx, ip, ifbEgress)
+		} else {
+			logger.G(ctx).WithError(err).Warning("Unable to find ifb egress, during deallocation")
+		}
 
-	ifbIngress, err := netlink.LinkByName(vpc.IngressIFB)
-	if err == nil {
-		removeClass(ctx, ip, ifbIngress)
-	} else {
-		logger.G(ctx).WithError(err).Warning("Unable to find ifb ingress, during deallocation")
+		ifbIngress, err := netlink.LinkByName(vpc.IngressIFB)
+		if err == nil {
+			removeClass(ctx, ip, ifbIngress)
+		} else {
+			logger.G(ctx).WithError(err).Warning("Unable to find ifb ingress, during deallocation")
+		}
 	}
 }
 
-func deleteLink(ctx context.Context, link netlink.Link, netnsfd int) {
+func deleteLink(ctx context.Context, link netlink.Link, netNS interface{}) {
 	if link == nil {
 		logger.G(ctx).Debug("Link not setup, not deleting link")
 		return
 	}
+	netnsfd, _, err := getNsFd(netNS)
```

Comment:
Can we get away with this `deleteLink` function remaining the same as before, just the caller gives it a different fd depending on which link it is deleting?